bbot/modules/internal/excavate.py

```diff
@@ -36,6 +36,24 @@ def report(self, result, name, event):
         pass
 
 
+class CSPExtractor(BaseExtractor):
+    regexes = {"CSP": r"(?i)(?m)Content-Security-Policy:.+$"}
+
+    def extract_domains(self, csp):
+        domains = dns_name_regex.findall(csp)
+        unique_domains = set(domains)
+        return unique_domains
+
+    async def search(self, content, event, **kwargs):
+        async for csp, name in self._search(content, event, **kwargs):
+            extracted_domains = self.extract_domains(csp)
+            for domain in extracted_domains:
+                self.report(domain, event, **kwargs)
+
+    def report(self, domain, event, **kwargs):
+        self.excavate.emit_event(domain, "DNS_NAME_UNRESOLVED", source=event)
```

Comment:
Instead of emitting as `DNS_NAME_UNRESOLVED` I would emit as a `DNS_NAME` with `tags=["affiliate"]`.